modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java

```diff
@@ -146,8 +149,7 @@
 
 import static org.elasticsearch.plugins.AnalysisPlugin.requiresAnalysisSettings;
 
-public class CommonAnalysisPlugin extends Plugin implements AnalysisPlugin, ScriptPlugin {
-
+public class CommonAnalysisPlugin extends Plugin implements AnalysisPlugin, ScriptPlugin, SystemIndexPlugin {
```

Comment:
Should we create a separate plugin that will include API as well, and separate from Analysis?

That would help encapsulating the new features, and would be cohesive in terms of the APIs plus the System Index they manage.

WDYT?

Reply:
Talked to @carlosdelest offline, and we decided that for now for simplicity we will keep this in `CommonAnalysisPlugin`. Later we may revisit this.